.github/workflows/release.yml

```diff
@@ -0,0 +1,97 @@
+name: Build and release
+on:
+    push:
+      tags:
+        - 'v[0-9]+\.[0-9]+\.[0-9]+'
+
+jobs:
+    build:
+        name: build release binaries on ${{ matrix.os }}
+        strategy:
+            fail-fast: false
+            matrix:
+                os: ["ubuntu-latest"]
+        runs-on: ${{ matrix.os }}
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+
+            - name: Install rust
+              id: rust
+              uses: dtolnay/rust-toolchain@stable
+
+            - name: Install cross
+              run: |
+                cargo install cross --git https://github.com/cross-rs/cross
+
+            - name: Build unix
+              id: unix_build
+              run: |
+                cross build --release --locked --target x86_64-unknown-linux-gnu
+                cross build --release --locked --target armv7-unknown-linux-gnueabihf
+                cross build --release --locked --target aarch64-unknown-linux-gnu
+                tar -czf "teos-${{github.ref_name}}-aarch64-linux-gnu.tar.gz" --transform 's|.*/||' "target/aarch64-unknown-linux-gnu/release/teosd" "target/aarch64-unknown-linux-gnu/release/teos-cli"
+                tar -czf "teos-${{github.ref_name}}-armv7-linux-gnueabihf.tar.gz" --transform 's|.*/||' "target/armv7-unknown-linux-gnueabihf/release/teosd" "target/armv7-unknown-linux-gnueabihf/release/teos-cli"
+                tar -czf "teos-${{github.ref_name}}-x86_64-linux-gnu.tar.gz" --transform 's|.*/||' "target/x86_64-unknown-linux-gnu/release/teosd" "target/x86_64-unknown-linux-gnu/release/teos-cli"
+                tar -czf "watchtower-client-${{github.ref_name}}-aarch64-linux-gnu.tar.gz" --transform 's|.*/||' "target/aarch64-unknown-linux-gnu/release/watchtower-client"
+                tar -czf "watchtower-client-${{github.ref_name}}-armv7-linux-gnueabihf.tar.gz" --transform 's|.*/||' "target/armv7-unknown-linux-gnueabihf/release/watchtower-client"
+                tar -czf "watchtower-client-${{github.ref_name}}-x86_64-linux-gnu.tar.gz" --transform 's|.*/||' "target/x86_64-unknown-linux-gnu/release/watchtower-client"
```

Comment:
Let's split this part into its own step for a little bit more readability.

Also, what do you think about removing the `gnu` & `gnueabihf` endings?

Reply:
Personally i don't care about these. I removed them.